OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -163,6 +169,20 @@ public Aircraft(ActorInitializer init, AircraftInfo info)
 			IsPlane = !info.CanHover;
 		}
 
+		public virtual IEnumerable<VariableObserver> GetVariableObservers()
+		{
+			if (Info.LandOnCondition != null)
+				yield return new VariableObserver(LandIfConditionChanged, Info.LandOnCondition.Variables);
+		}
+
+		void LandIfConditionChanged(Actor self, IReadOnlyDictionary<string, int> conditions)
+		{
+			if (Info.LandOnCondition == null)
```

Comment:
I don't see any way for this to be called if `Info.LandOnCondition` is null, so this check should be useless.